sdk-ext-lib/src/main/java/cash/z/ecc/sdk/model/WalletAddresses.kt

```diff
@@ -2,44 +2,47 @@ package cash.z.ecc.sdk.model
 
 import cash.z.ecc.android.bip39.Mnemonics
 import cash.z.ecc.android.bip39.toSeed
+import cash.z.ecc.android.sdk.Synchronizer
+import cash.z.ecc.android.sdk.model.Account
 import cash.z.ecc.android.sdk.tool.DerivationTool
+import cash.z.ecc.android.sdk.type.UnifiedFullViewingKey
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 
 data class WalletAddresses(
     val unified: WalletAddress.Unified,
     val shieldedSapling: WalletAddress.ShieldedSapling,
     val transparent: WalletAddress.Transparent,
-    val viewingKey: String
+    val viewingKey: UnifiedFullViewingKey
 ) {
     // Override to prevent leaking details in logs
     override fun toString() = "WalletAddresses"
 
     companion object {
-        suspend fun new(persistableWallet: PersistableWallet): WalletAddresses {
+        suspend fun new(persistableWallet: PersistableWallet, synchronizer: Synchronizer): WalletAddresses {
             // Dispatcher needed because SecureRandom is loaded, which is slow and performs IO
             // https://github.com/zcash/kotlin-bip39/issues/13
             val bip39Seed = withContext(Dispatchers.IO) {
                 Mnemonics.MnemonicCode(persistableWallet.seedPhrase.joinToString()).toSeed()
             }
 
-            val viewingKey = DerivationTool.deriveUnifiedViewingKeys(bip39Seed, persistableWallet.network)[0].extpub
+            val viewingKey = DerivationTool.deriveUnifiedFullViewingKeys(bip39Seed, persistableWallet.network, 1)[0]
```

Comment:
@str4d Am I remembering correctly that we don't want to show viewing keys to users anymore?